Tests/MessagingPush/MessagingPushImplementationTest.swift

```diff
@@ -122,4 +125,41 @@ class MessagingPushImplementationTest: UnitTest {
         XCTAssertEqual(actualQueueTaskData.event, givenEvent)
         XCTAssertEqual(actualQueueTaskData.deviceToken, givenDeviceToken)
     }
+    
+    // MARK: DeviceAttributes
+    func test_deviceAttributes_givenDeviceToken_expectDefaultDeviceAttributesDictionary() {
+        let givenDeviceToken = String.random
+        var config = SdkConfig()
+        config.autoTrackDeviceAttributes = true
+        sdkConfigStoreMock.config = config
+        #if canImport(UIKit)
+        messagingPush.deviceAttributes(deviceToken: givenDeviceToken) { defaultAttributes in
+            let expectedAppVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+            let expectedOS = "\(UIDevice.current.systemName) \(UIDevice.current.systemVersion)"
+            let expectedLocale = DeviceInfo.deviceLocale.value.replacingOccurrences(of: "_", with: "-")
+            let expectedModel = UIDevice.deviceModelCode
+            let expectedSDKVersion = SdkVersion.version
+            XCTAssertEqual(defaultAttributes!["appVersion"], expectedAppVersion)
+            XCTAssertEqual(defaultAttributes!["deviceOs"], expectedOS)
+            XCTAssertEqual(defaultAttributes!["deviceLocale"], expectedLocale)
+            XCTAssertEqual(defaultAttributes!["deviceModel"], expectedModel)
+            XCTAssertEqual(defaultAttributes!["pushSubscribed"], "false")
+            XCTAssertEqual(defaultAttributes!["cioSdkVersion"], expectedSDKVersion)
+        }
+        #endif
+    }
+    
+    func test_deviceAttributes_givenDeviceToken_expectNil() {
+        let givenDeviceToken = String.random
+        var config = SdkConfig()
+        config.autoTrackDeviceAttributes = false
+        sdkConfigStoreMock.config = config
+
+        // This function returns `nil` in case
+        // `autoTrackDeviceAttributes` is set to `false`
+        messagingPush.deviceAttributes(deviceToken: givenDeviceToken) { defaultAttributes in
+            
+            XCTAssertNil(defaultAttributes)
+        }
```

Comment:
Since this code is async, you will need [your test function to wait for the async call to finish](https://github.com/customerio/customerio-ios/blob/8716b3391f2de135cd7f6a42950b81174e1be39b/Tests/Tracking/Service/HttpClientTest.swift#L53-L63)

Reply:
✅